src/security/cert-storage.ts

```diff
@@ -135,17 +136,92 @@ export class CertStorage implements SecurityAgent {
     storage: Storage,
     fw: Forwarder,
     prvKeyBits: Uint8Array,
-    interestLifetime = 5000,
   ) {
     const result = new CertStorage(
       trustAnchor,
       ownCertificate,
       storage,
       fw,
       prvKeyBits,
-      interestLifetime,
     );
     await result.readyEvent;
     return result;
   }
+
+  /**
+   * Modified version of getCertificate function, designed for trying to
+   * fetch PoR from all other already trusted parties
+   * @param strangerKeyname First part of PoR name, i.e. "/<𝑍𝑏 name>/KEY/<𝑇𝑏 keyid>"
+   */
+  async fetchPoR(strangerKeyname: string) {
+    // loop through trusted certificates
+    for (const item of this.trustedNames) {
+      // Construct <Z_a name encoded>
+      const parts: string[] = item.split('/');
+      let partsBeforeKey: string = '';
+      for (let i = 0; i < parts.length; i++) { //put <Z_a name encoded> together with a loop
+        if (parts[i].includes('KEY')) {
+          break;
+        }
+        partsBeforeKey += parts[i] + '/';
+      }
+
+      // return if we found that one of the users we trust, also trusts the new user
+      const response: Certificate | undefined = await this.getCertificate(
+        new Name(strangerKeyname + partsBeforeKey + 'v1'),
+        false,
```

Comment:
This means PoR can be recursively signed by other unknown certifcates. Please check if the behavior if desired.